File: iOS/BettingHub/BettingHub/Source/Models/Sport.swift
//
//  Sport.swift
//  BettingHub
//
//  Created by Maxim Bezdenezhnykh on 21.04.2020.
//  Copyright © 2020 Maxim Bezdenezhnykh. All rights reserved.
//

enum Sport: Int {
    case all = 0
    case football = 1
    case tennis = 2
    case basketball = 3
    case hockey = 4
    case other = 5
    
    var sportName: String {
        [
            .all: "all",
            .football: "football",
            .tennis: "tennis",
            .basketball: "basketball",
            .hockey: "hockey",
            .other: "other"
        ][self]!
    }
    
    static func getAll() -> [Sport] {
        return [.all, .football, .tennis, .basketball, .hockey, .other]
    }
}

File: android/BettingHUB/app/src/main/java/com/xbethub/webview/enums/Sport.kt
package com.xbethub.webview.enums

import com.xbethub.webview.R

enum class Sport(val stringRes: Int, val backendValue: String) {
    ALL(R.string.allSports, "all")
    , FOOTBALL(R.string.football, "1")
    , TENNIS(R.string.tennis, "2")
    , BASKETBALL(R.string.basketball, "3")
    , HOCKEY(R.string.hockey, "4")
    , OTHER(R.string.other, "5")
}
